docs/manual/map-key-checker.tex

```diff
@@ -49,6 +49,19 @@ \section{Map key annotations\label{map-key-annotations}}
 \label{fig-map-key-keyfor-hierarchy}
 \end{figure}
 
+\section{Default annotations\label{map-key-defaults}}
+
+The null literal is implicitly annotated with \code{@UnknownKeyFor}, because
+\code{null} might be a key for any map.
+
+Lower bounds are defaulted to \code{@KeyForBottom}.
+This reduces the number of false
+positives, but it does require you to take action if you expect a
+type variable to be instantiated with \code{@KeyFor} for a particular map.
+In that case,
+you must explicitly annotate the lower bound of that type variable with
+\code{@KeyForBottom}.  For example, \code{List<@KeyForBottom T>}.
```

Comment:
Can you expand this into a complete example?  This doesn't show a full use or clarify the question about whether to write at a declaration or use site.  I think an example will help to clarify the issue.